File: SRM 697/DivisibleSetDiv1.cs
using System;
using System.IO;
using System.Collections.Generic;
using System.Text;

public class DivisibleSetDiv1 {
	public string isPossible(int[] b) {
        Array.Sort(b);
        Array.Reverse(b);
        int[] a = new int[b.Length];
        for (int i = 0; i < a.Length; ++i) {
            a[i] = i + 1;
        }
        for (int i = 0; i < 100000; ++i) {
            bool flag = true;
            for (int j = 0; j < b.Length; ++j) {
                int sum = 0;
                int need = a[j] * b[j];
                for (int k = 0; k < b.Length; ++k) {
                    if (j == k) continue;
                    sum += a[k];
                }
                if (need < sum) {
                    for (int k = j; k < b.Length; ++k) {
                        a[k]++;
                        flag = false;
                    }
                    break;
                }
            }
            if (flag) {
                return "Possible";
            }
        }
		return "Impossible";
	}

// CUT begin
	static bool DoTest(int[] b, string __expected) {
		DateTime startTime = DateTime.Now;
		Exception exception = null;
		DivisibleSetDiv1 instance = new DivisibleSetDiv1();
		string __result = "";
		try {
			__result = instance.isPossible(b);
		}
		catch (Exception e) { exception = e; }
		TimeSpan __elapsed = new TimeSpan(DateTime.Now.Ticks - startTime.Ticks);

		if (exception != null) {
			Console.Error.WriteLine("RUNTIME ERROR!");
			Console.Error.WriteLine(exception);
			return false;
		}
		else if (__result == __expected) {
			Console.Error.WriteLine("PASSED! " + string.Format("({0:0.00} seconds)", __elapsed.TotalSeconds));
			return true;
		}
		else {
			Console.Error.WriteLine("FAILED! " + string.Format("({0:0.00} seconds)", __elapsed.TotalSeconds));
			Console.Error.WriteLine("           Expected: " + "\"" + __expected + "\"");
			Console.Error.WriteLine("           Received: " + "\"" + __result + "\"");
			return false;
		}
	}

	public static void Main(string[] args){
		HashSet<int> cases = new HashSet<int>();
        for (int i = 0; i < args.Length; ++i) cases.Add(int.Parse(args[i]));

		Console.Error.WriteLine("DivisibleSetDiv1 (300 Points)");
		Console.Error.WriteLine();

    	int nCases = 0, nPassed = 0;
    	using (var reader = File.OpenText("DivisibleSetDiv1.sample")) {
            while (true) {
                string line = reader.ReadLine();
                if (line == null || !line.StartsWith("--"))
                    break;

                int[] b = new int[int.Parse(reader.ReadLine())];
                for (int i = 0; i < b.Length; ++i)
                    b[i] = (int) Convert.ChangeType(reader.ReadLine(), typeof(int));
                reader.ReadLine();
                string __answer = (string) Convert.ChangeType(reader.ReadLine(), typeof(string));

                nCases++;
                if (cases.Count > 0 && !cases.Contains(nCases - 1))
                	continue;
                Console.Error.Write(string.Format("  Testcase #{0} ... ", nCases - 1));
                if (DoTest(b, __answer))
                    nPassed++;
            }
    	}

        if (cases.Count > 0) nCases = cases.Count;
		Console.Error.WriteLine();
		Console.Error.WriteLine(string.Format("Passed : {0}/{1} cases", nPassed, nCases));

		DateTime Jan1st1970 = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
		int T = (int)((DateTime.UtcNow - Jan1st1970).TotalSeconds - 1470593751);
		double PT = T / 60.0, TT = 75.0;
		Console.Error.WriteLine(string.Format("Time   : {0} minutes {1} secs", T / 60, T % 60));
		Console.Error.WriteLine(string.Format("Score  : {0:0.00} points", 300 * (0.3 + (0.7 * TT * TT) / (10.0 * PT * PT + TT * TT))));
	}
// CUT end
}

File: SRM 697/DivisibleSetDiv1.java
import java.io.*;
import java.math.BigInteger;
import java.util.*;

public class DivisibleSetDiv1 {
	public String isPossible(int[] b) {
		BigInteger bigInteger = BigInteger.ONE;
		for (int x : b) {
			bigInteger = bigInteger.multiply(BigInteger.valueOf(x + 1));
		}
		BigInteger num = BigInteger.ZERO;
		for (int x : b) {
			num = num.add(bigInteger.divide(BigInteger.valueOf(x + 1)));
		}
		int cmp = num.compareTo(bigInteger);
		if (cmp > 0) {
			return "Impossible";
		}
		if (cmp < 0) {
			return "Possible";
		}
		for (int i = 0; i < b.length; ++i) {

			for (int j = i + 1; j < b.length; ++j) {

				if (b[i] == b[j]) {

					return "Impossible";
				}
			}
		}
		return "Possible";
	}

// CUT begin
	public static void main(String[] args){
		System.err.println("DivisibleSetDiv1 (300 Points)");
		System.err.println();
		HashSet<Integer> cases = new HashSet<Integer>();
        for (int i = 0; i < args.length; ++i) cases.add(Integer.parseInt(args[i]));
        runTest(cases);
	}

	static void runTest(HashSet<Integer> caseSet) {
	    int cases = 0, passed = 0;
	    while (true) {
	    	String label = Reader.nextLine();
	    	if (label == null || !label.startsWith("--"))
	    		break;

            int[] b = new int[Integer.parseInt(Reader.nextLine())];
            for (int i = 0; i < b.length; ++i)
                b[i] = Integer.parseInt(Reader.nextLine());
            Reader.nextLine();
            String __answer = Reader.nextLine();

            cases++;
            if (caseSet.size() > 0 && !caseSet.contains(cases - 1))
                continue;
    		System.err.print(String.format("  Testcase #%d ... ", cases - 1));

            if (doTest(b, __answer))
                passed++;
	    }
	    if (caseSet.size() > 0) cases = caseSet.size();
        System.err.println(String.format("%nPassed : %d/%d cases", passed, cases));

        int T = (int)(System.currentTimeMillis() / 1000) - 1470680918;
        double PT = T / 60.0, TT = 75.0;
        System.err.println(String.format("Time   : %d minutes %d secs%nScore  : %.2f points", T / 60, T % 60, 300 * (0.3 + (0.7 * TT * TT) / (10.0 * PT * PT + TT * TT))));
	}

	static boolean doTest(int[] b, String __expected) {
		long startTime = System.currentTimeMillis();
		Throwable exception = null;
		DivisibleSetDiv1 instance = new DivisibleSetDiv1();
		String __result = "";
		try {
			__result = instance.isPossible(b);
		}
		catch (Throwable e) { exception = e; }
		double elapsed = (System.currentTimeMillis() - startTime) / 1000.0;

		if (exception != null) {
			System.err.println("RUNTIME ERROR!");
			exception.printStackTrace();
			return false;
		}
		else if (__expected.equals(__result)) {
			System.err.println("PASSED! " + String.format("(%.2f seconds)", elapsed));
			return true;
		}
		else {
			System.err.println("FAILED! " + String.format("(%.2f seconds)", elapsed));
			System.err.println("           Expected: " + "\"" + __expected + "\"");
			System.err.println("           Received: " + "\"" + __result + "\"");
			return false;
		}
	}

	static class Reader {
        private static final String dataFileName = "/Users/resry/topcoder/SRM 697/DivisibleSetDiv1.sample";
	    private static BufferedReader reader;

	    public static String nextLine() {
	        try {
                if (reader == null) {
                    reader = new BufferedReader(new InputStreamReader(new FileInputStream(dataFileName)));
                }
                return reader.readLine();
	        }
	        catch (IOException e) {
	            System.err.println("FATAL!! IOException");
	            e.printStackTrace();
	            System.exit(1);
	        }
	        return "";
	    }
	}
// CUT end
}
